### PR TITLE
New Sphinx Theme

### PR DESCRIPTION
**Pull request by robcohen**
Sphinx theme to match what is being used on the other sites.

### CODE REVIEW

File: doc/.sphinx/_templates/layout.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% extends "!layout.html" %}
+
+{% block footer %}
+{{ super() }}
+<div class="footer">YEP THIS IS AN EXAMPLE
```
Comment (michaelpj) [inline doc/.sphinx/_templates/layout.html:5]: Is this supposed to say this?

File: doc/conf.py [hunk 2]
```diff
@@ -1,48 +1,52 @@
...
-# sys.path.insert(0, os.path.abspath('.'))
+import sys
+import os
+import sphinx_rtd_theme
+import recommonmark
+
+from recommonmark.transform import AutoStructify
+from os.path import abspath, join, dirname
+
+sys.path.insert(0, abspath(join(dirname(__file__))))
```
Comment (michaelpj) [inline doc/conf.py:11]: What's this for?

File: doc/conf.py [hunk 3]
```diff
@@ -1,48 +1,52 @@
...
+
 extensions = [
-  'hs_domain',
-  'sphinx.ext.intersphinx'
+    "sphinx_rtd_theme",
+    'recommonmark',
+    'sphinx_markdown_tables',
+    'sphinxemoji.sphinxemoji',
+    "sphinx.ext.autodoc",
+    "sphinx.ext.autosummary",
```
Comment (michaelpj) [inline doc/conf.py:44]: autodoc and autosummary are for python code, we don't need them

File: doc/conf.py [hunk 4]
```diff
@@ -1,48 +1,52 @@
...
+    "sphinx.ext.intersphinx",
+    "sphinx.ext.viewcode",
```
Comment (michaelpj) [inline doc/conf.py:46]: This is also python-specific

File: doc/conf.py [hunk 5]
```diff
@@ -57,14 +61,71 @@
...
+templates_path = ['.sphinx/_templates']
+html_static_path = ['.sphinx/_static']
+
+source_suffix = {
+    '.rst': 'restructuredtext',
+    '.md': 'markdown',
+}
+
+intersphinx_mapping = {
+    "commandsv1": (
```
Comment (michaelpj) [inline doc/conf.py:75]: Why are we linking to the docs for FIRST? Also this overrides the mapping set earlier

File: doc/conf.py [hunk 6]
```diff
@@ -57,14 +61,71 @@
...
+    'collapse_navigation': True,
+    'sticky_navigation': True,
+    'navigation_depth': 4,
+    'includehidden': True,
+    'titles_only': False
+}
 
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
```
Comment (michaelpj) [inline doc/conf.py:111]: This comment is in the wrong place now.

File: doc/conf.py [hunk 7]
```diff
@@ -57,14 +61,71 @@
...
-html_static_path = []
+
+html_logo = ".sphinx/cardano-logo.png"
+
+html_context = {
+  "display_github": True, # Add 'Edit on Github' link instead of 'View page source'
+  "github_user": "input-output-hk",
+  "github_repo": "cardano-node",
```
Comment (michaelpj) [inline doc/conf.py:118]: This is definitely isn't right, not sure about the other arguments.

File: .readthedocs.yml [hunk 8]
```diff
@@ -1,15 +1,11 @@
...
-# Build documentation in the docs/ directory with Sphinx
 sphinx:
   configuration: doc/conf.py
 
+# Optionally set the version of Python and requirements required to build your docs
 python:
   version: 3.7
   install:
-    - requirements: doc/requirements.txt
+    - requirements: doc/.sphinx/requirements.txt
```
Comment (michaelpj) [inline .readthedocs.yml:11]: I don't especially mind, but what was wrong with the old location? It's the requirements for building the Python project rooted in `doc`, it seems reasonable to have them there.

**Comment by michaelpj** [comment]
I also need to fix up some Nix stuff, I'll just push to your branch.

File: .readthedocs.yml (hunk 8 above)
Comment (michaelpj) [reply .readthedocs.yml:11]: Similarly for the templates etc. I guess you're trying to hide the sphinx stuff from people? Not worth it IMO but I don't mind being consistent.

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,29 @@
+###############
```
Comment (michaelpj) [inline README.rst:1]: I hadn't noticed you'd moved this to the root. Please don't do this: we have a perfectly good readme for the repository already.

**Comment by michaelpj** [comment]
I went ahead and addressed a bunch of the minor comments since I was working on the branch.

File: doc/conf.py (hunk 6 above)
Comment (michaelpj) [reply doc/conf.py]: Removed.

File: doc/conf.py (hunk 5 above)
Comment (michaelpj) [reply doc/conf.py]: Removed.

File: doc/conf.py (hunk 3 above)
Comment (michaelpj) [reply doc/conf.py]: Removed.

File: doc/conf.py (hunk 4 above)
Comment (michaelpj) [reply doc/conf.py]: Removed

File: README.rst (hunk 9 above)
Comment (michaelpj) [reply README.rst]: Reverted.

File: doc/conf.py (hunk 7 above)
Comment (michaelpj) [reply doc/conf.py]: Updated to what I think are the right arguments.

File: doc/.sphinx/_templates/layout.html (hunk 1 above)
Comment (robcohen) [reply doc/.sphinx/_templates/layout.html:5]: Nope, but it doesn't render. I'll fix it though.

File: doc/conf.py (hunk 2 above)
Comment (robcohen) [reply doc/conf.py:11]: recommonmark is to parse markdown.

File: .readthedocs.yml (hunk 8 above)
Comment (robcohen) [reply .readthedocs.yml:11]: Yeah I was trying to hide as much as possible. I tried to hide conf.py too, but that didn't see possible, something to do with the way the python config references things in a path. It won't allow `../something`.

Comment (robcohen) [reply .readthedocs.yml:11]: Don't particularly care about consistency of location of requirements, happy to leave that to the owner of the repo. So long as everything except README.rst is under /doc

**Comment by robcohen** [comment]
@michaelpj removed the file under _template, other than that looks good to me.

File: doc/conf.py (hunk 2 above)
Comment (michaelpj) [reply doc/conf.py:11]: I meant the bit where we add the current directory to the python path.

File: .readthedocs.yml (hunk 8 above)
Comment (michaelpj) [reply .readthedocs.yml:11]: FWIW I don't feel the need to hide that in our case, I'd actually rather have it be more visible. So I think I'm going to put it back if you don't mind. Shouldn't affect anything else.

**Comment by michaelpj** [comment]
Thanks Rob!

**Pull request merged**